core/src/main/java/com/linecorp/armeria/common/logging/RequestLogAccess.java

```diff
@@ -260,4 +263,10 @@ public interface RequestLogAccess {
      * ordered by the time they were added.
      */
     List<RequestLogAccess> children();
+
+    /**
+     * Returns all available {@link RequestLogProperty}s.
+     */
+    @UnstableApi
+    Set<RequestLogProperty> availableProperties();
```

Comment:
I'd like to ask you to remove this method in favor of `RequestLogAccess.isAvailable()` and/or `availabilityStamp()` because of the potential performance penalty this will bring to a user.

If you need to check the availability of multiple properties, you can use `availabilityStamp()` instead: (Note that `availabilityStamp()` incurs a volatile read.)

```
int stamp = log.availabilityStamp();
if (stamp & REQUEST_START_TIME.flag() != 0) {
  // REQUEST_START_TIME is available.
}
if (stamp & RESPONSE_START_TIME.flag() == 0) {
  // RESPONSE_START_TIME is NOT available.
}
```

If you're concerned about the verbosity, how about adding `RequestLogProperty.isAvailable(int availabilityStamp)`?

```
int stamp = log.availabilityStamp();
if (REQUEST_START_TIME.isAvailable(stamp)) {
  ...
}
if (!RESPONSE_START_TIME.isAvailable(stamp)) {
  ...
}
```